src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -124,6 +160,28 @@ CHIP_ERROR AES_CCM_decrypt(const uint8_t * ciphertext, size_t ciphertext_length,
     VerifyOrReturnError(aad != nullptr || aad_length == 0, CHIP_ERROR_INVALID_ARGUMENT);
 
     const psa_algorithm_t algorithm = PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM, tag_length);
+
+#ifdef CHIP_CRYPTO_PSA_AEAD_SINGLE_PART
+    // Single-part API with static buffer
+    uint8_t temp_buf[PSA_AEAD_TEMP_BUFFER_SIZE];
+    VerifyOrReturnError(ciphertext_length + tag_length <= PSA_AEAD_TEMP_BUFFER_SIZE,
+                        CHIP_ERROR_INVALID_ARGUMENT);
+
+    memcpy(temp_buf, ciphertext, ciphertext_length);
+    memcpy(temp_buf + ciphertext_length, tag, tag_length);
+
+    size_t output_len = 0;
+    psa_status_t status = psa_aead_decrypt(key.As<psa_key_id_t>(), algorithm,
+                                          nonce, nonce_length,
+                                          aad, aad_length,
+                                          temp_buf, ciphertext_length + tag_length,
+                                          plaintext, ciphertext_length,
+                                          &output_len);
+    VerifyOrReturnError(status == PSA_SUCCESS && output_len == ciphertext_length,
+                        (status == PSA_ERROR_INVALID_SIGNATURE)
```

Comment:
nit: but it seems for multi-part we don't return `CHIP_ERROR_INVALID_SIGNATURE`, neither do other backends.